Messaging.md

```diff
@@ -363,92 +371,210 @@ Witnesses should be sorted by the `input_serial_id` sent in `funding_input` defi
+* MUST reject if `announcement_public_key == attestation_public_key`.
+
+
+### The `proof_of_knowledge` Type
+
+A proof of knowledge is used to prove that oracles know the pre-images of the nonces they are using to prevent cancellation attacks in multi-oracle settings.
+
+#### `schnorr_proof_of_knowledge`
+
+1. implements: `proof_of_knowledge`
+1. type: 0
+1. data:
+   * [`signature`:`attestation_public_key_proof`]
+   * [`nb_signatures`:`bigsize`]
+   * [`nb_signatures*signature`:`nonce_proofs`]
+
+where each signature is generated over the message `dlcoraclepok`.
+
+### The `oracle_schemes` Type
+
+In order to enable oracles to support multiple attestation schemes, and clients to chose their preferred one, the `oracle_schemes` type can contain multiple attestations schemes.
+Each scheme is defined as a TLV so that clients can easily ignore schemes that they don't know about.
+Note that at the moment the only defined scheme is the `schnorr_scheme` that uses Schnorr signatures.
+
+#### `oracle_schemes`
+
+1. data:
+   * [`bigsize`:`nb_schemes`]
+   * [`nb_schemes*scheme`:`attestation_schemes`]
+
+#### `schnorr_scheme`
+
+1. type: 0
+1. data:
+   * [`x_point`:`attestation_public_key`]
+   * [`bigsize`:`nb_nonces`]
+   * [`nb_nonces*x_point`:`oracle_nonces`]
+   * [`proof_of_knowledge`:`proof_of_knowledge`]
+
+where `proof_of_knowledge` ensures that the oracle knows the pre-image of the `attestation_public_key` and all `oracle_nonce`.
+
+#### Requirements
+
+A node *MUST* validate the proof of knowledge in the case of multi oracle contracts.
+A node *MUST* discard any announcement failing this verification.
+
+#### Rationale
+
+While using an aggregate signature increases the complexity of oracle implementation and management, providing a signature per nonce would lead to a large number of signatures in particular for oracle attesting to numerical event.
+
 ### The `oracle_announcement` Type
 
 This type contains an `oracle_event` and a signature certifying its origination.
 As oracle announcements can be broadcast directly, they are encoded as [wire messages](#wire-messages).
 See [the Oracle specifications](./Oracle.md#oracle-announcements) for more details.
 
-**For backward compatibility reasons, this type is currently serialized as a TLV and uses u16 for collection prefix. It is expected to be changed to the new serialization format in a near future.**
-
 #### `oracle_announcement`
 
-1. type: 55332
-1. data:
-   * [`signature`:`annoucement_signature`]
-   * [`x_point`:`oracle_public_key`]
+1. type: 55354 (`oracle_announcement`)
+2. data:
+   * [`signature`:`announcement_signature`]
+   * [`oracle_metadata`:`oracle_metadata`]
    * [`oracle_event`:`oracle_event`]
 
-where `signature` is a Schnorr signature over a sha256 hash of the serialized `oracle_event`, using the tag `announcement/v0`.
+where both the `announcement_signature` is a Schnorr signature over a sha256 hash of the serialized `oracle_event`, using the tag `announcement/v1`.
```

Comment:
what is the public key used to check against the announcement signature?

Reply:
Clarified (it's the oracle_annoucement_public_key`